url8f84d8d956_klapaukh_JTuttePoly_tgz-pJ8-BitTutteJ8/src/graph/Util.java

```diff
@@ -0,0 +1,210 @@
+package graph;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+
+import util.Pair;
+
+// NOTES: only works for connected graphs at the moment
+
+public final class Util {
+	
+	public static boolean isomorphic(Graph g1, Graph g2) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 40-70 into a separate method.